libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -64,32 +64,45 @@ AP_GPS_SBF::AP_GPS_SBF(AP_GPS &_gps, AP_GPS::GPS_State &_state,
 bool
 AP_GPS_SBF::read(void)
 {
-    uint32_t now = AP_HAL::millis();
-
-    if (gps._auto_config != AP_GPS::GPS_AUTO_CONFIG_DISABLE &&
-        _init_blob_index < ARRAY_SIZE(_initialisation_blob)) {
-        const char *init_str = _initialisation_blob[_init_blob_index];
-
-        if (now > _init_blob_time) {
-            if (now > _config_last_ack_time + 2500) {
-                // try to enable input on the GPS port if we have not made progress on configuring it
-                Debug("SBF Sending port enable");
-                port->write((const uint8_t*)_port_enable, strlen(_port_enable));
-                _config_last_ack_time = now;
-            } else {
-                Debug("SBF sending init string: %s", init_str);
-                port->write((const uint8_t*)init_str, strlen(init_str));
-            }
-            _init_blob_time = now + 1000;
-        }
-    }
-
     bool ret = false;
-    while (port->available() > 0) {
+    uint32_t available_bytes = port->available();
+    for (uint32_t i = 0; i < available_bytes; i++) {
         uint8_t temp = port->read();
         ret |= parse(temp);
     }
 
+    if (gps._auto_config != AP_GPS::GPS_AUTO_CONFIG_DISABLE) {
+        if (_init_blob_index < ARRAY_SIZE(_initialisation_blob)) {
+            uint32_t now = AP_HAL::millis();
+            const char *init_str = _initialisation_blob[_init_blob_index];
+
+            if (now > _init_blob_time) {
+                if (now > _config_last_ack_time + 2500) {
+                    // try to enable input on the GPS port if we have not made progress on configuring it
+                    Debug("SBF Sending port enable");
+                    port->write((const uint8_t*)_port_enable, strlen(_port_enable));
+                    _config_last_ack_time = now;
+                } else {
+                    Debug("SBF sending init string: %s", init_str);
+                    port->write((const uint8_t*)init_str, strlen(init_str));
+                }
+                _init_blob_time = now + 1000;
+            }
+        } else if (gps._raw_data >= 2) { // only manage disarm/rearms when the user opts into it
```

Comment:
Should we make this match exactly 2? I'm thinking that someone might leave this at 5 from a previous uBlox use.

Reply:
Made the change, I've seen either approach used elsewhere, but don't feel terribly strongly about it :)